### ISSUE TITLE
The Slime Critter attack throws the player several meters away

### ISSUE THREAD

**Issue by Eddrico** [UnityTechnologies/open-project-1#349]
Short description
When the slime critter attacks and the player jumps, the inertia of the attack is transferred to the player, throwing the player several meters away.

Expected behavior
The attack should not push the player so far.

To Reproduce

    1.- Open a scene with the TestingGround. Press play.
    2.- Get near a Slime Critter and wait for his attack.
    3.- When the attack is about to connect, press the jump button.
    4.- Observe that the player is sent flying away.

Notes

    Here's a video of the issue:

[https://www.youtube.com/watch?v=pnKQh3bKC8Y](url)

**Comment by ciro-unity** [comment]
Forum link: https://forum.unity.com/threads/the-slime-critter-attack-throws-the-player-several-meters-away.1050692/

**Comment by treivize** [comment]
For this one, I would wait for the coming PR about combat system to be merged. Jumping will be prevent when the player is getting hit by the slime critter attack, so this bug should not be reproduceable after the integration of this part of the system.

**Comment by AlexandreGheraibia** [comment]
In SlimeCritterAttackController, is it normal that the difference between positions isn't normalized ? 
 `_propelTargetVector = (_playerTransform.Transform.position - transform.position) * _propelFactor / _propelDuration;`

**Comment by ciro-unity** [comment]
The issue doesn't happen because the player jumps. It just happens when it's squashed by the attack against a rock or wall.

I actually made a video of it some time ago:

https://youtu.be/aQ9KuTuTNbM

I'm not jumping at all here!

**Comment by treivize** [comment]
Indeed, but are you able to reproduce it now that the player is really getting hit by the critter? I have just tried multiple times to reproduce it without success.

**Comment by ciro-unity** [comment]
No I don't think so. The initial description can throw you off guard: it's not that the inertia is transferred (I think) since the player has no Rigidbody. Most probably what was happening is that the Critter would move into the player -> Next frame the player is pushed out (as would two colliders one inside the other).
But now that the Hit is implemented, the player instantly plays an animation which keeps it anchored to the spot it is (root motion?) so it doesn't fly away.  Why doesn't he fly away after the animation is done, I'm not sure.
(these are all guesses)

In any case, no, I don't generally see it now. I will check again for some time, then close.

**Comment by AlexandreGheraibia** [comment]
perhaps because in HasHitHeadCondition you have the following code : 
bool isMovingUpwards = _protagonistScript.movementVector.y > 0f;
		if (isMovingUpwards)
		{
			// Making sure the collision is near the top of the head
			float permittedDistance = _characterController.radius / 2f;
			float topPositionY = _transform.position.y + _characterController.height;
			float distance = Mathf.Abs(_protagonistScript.lastHit.point.y - topPositionY);
			if (distance <= permittedDistance)
			{
				_protagonistScript.jumpInput = false;
				**_protagonistScript.movementVector.y = 0f;**

				return true;
			}
		}

**Comment by AlexandreGheraibia** [comment]
velocity ​​is zero when you enter idle state

**Comment by r4hulCorleone** [comment]
Hi, as i informed in unity forum, I like to work on this bug, producing the bug was hard for me because i need better timing, so i wrote a simple code that player does a jump action with delay after critter starts to attack, and with a proper delay, i reproduced the bug when player jumps successfully for a lot of times. I will search for the root of this bug, i think  @AlexandreGheraibia comment is a good start point for me. i will notify you when i find out more about the bug.

**Comment by paradox-mp** [comment]
I think this bug is still active.  I can reproduce a minor version of it.  I've included two examples below.  Here's a short description.
The player is not able to walk up the smaller rocks,  However, when the slime attacks hit the player, the player is pushed on top of the rocks.

This part is pure speculation...
It seems similar to the bug described above which is why I posted it here.  If @ciro-unity want me to open a new bug report for this issue, I will do so.
